From patchwork Thu May  6 13:37:50 2021
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 8bit
X-Patchwork-Submitter: =?utf-8?q?Philippe_Mathieu-Daud=C3=A9?=
 <philmd@redhat.com>
X-Patchwork-Id: 12242145
Return-Path: <kvm-owner@kernel.org>
X-Spam-Checker-Version: SpamAssassin 3.4.0 (2014-02-07) on
	aws-us-west-2-korg-lkml-1.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No, score=-19.4 required=3.0 tests=BAYES_00,DKIMWL_WL_HIGH,
	DKIM_SIGNED,DKIM_VALID,DKIM_VALID_AU,HEADER_FROM_DIFFERENT_DOMAINS,
	INCLUDES_CR_TRAILER,INCLUDES_PATCH,MAILING_LIST_MULTI,SPF_HELO_NONE,SPF_PASS,
	USER_AGENT_GIT autolearn=ham autolearn_force=no version=3.4.0
Received: from mail.kernel.org (mail.kernel.org [198.145.29.99])
	by smtp.lore.kernel.org (Postfix) with ESMTP id 96BF3C433B4
	for <kvm@archiver.kernel.org>; Thu,  6 May 2021 13:38:09 +0000 (UTC)
Received: from vger.kernel.org (vger.kernel.org [23.128.96.18])
	by mail.kernel.org (Postfix) with ESMTP id 5EB0D613F9
	for <kvm@archiver.kernel.org>; Thu,  6 May 2021 13:38:09 +0000 (UTC)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S234310AbhEFNjG (ORCPT <rfc822;kvm@archiver.kernel.org>);
        Thu, 6 May 2021 09:39:06 -0400
Received: from us-smtp-delivery-124.mimecast.com ([170.10.133.124]:30329 "EHLO
        us-smtp-delivery-124.mimecast.com" rhost-flags-OK-OK-OK-OK)
        by vger.kernel.org with ESMTP id S234072AbhEFNjG (ORCPT
        <rfc822;kvm@vger.kernel.org>); Thu, 6 May 2021 09:39:06 -0400
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=redhat.com;
        s=mimecast20190719; t=1620308288;
        h=from:from:reply-to:subject:subject:date:date:message-id:message-id:
         to:to:cc:cc:mime-version:mime-version:content-type:content-type:
         content-transfer-encoding:content-transfer-encoding:
         in-reply-to:in-reply-to:references:references;
        bh=/bdOhY3sBOcIHJeoCe6CkJ1QfZvAaT12hvfPxmFulM0=;
        b=diAk3LMOlfbES4tV7FjKNoalRYDIzxql+fuvw50YYXcjrxxjmW98+zMC6uRAF5HQrsAU7U
        X1MzwlvgmJLId0zYbCZutsDUbsq7JH3SjRoBb3iSDlRZljwSgcpEUpuuV/CUbXl3uJW9Ms
        gyNv0PZMEjVVs+XkW90LB4grNr5xT/4=
Received: from mail-wr1-f72.google.com (mail-wr1-f72.google.com
 [209.85.221.72]) (Using TLS) by relay.mimecast.com with ESMTP id
 us-mta-190-IUzylFh0Nz-lohi4v2qUBQ-1; Thu, 06 May 2021 09:38:06 -0400
X-MC-Unique: IUzylFh0Nz-lohi4v2qUBQ-1
Received: by mail-wr1-f72.google.com with SMTP id
 93-20020adf80e60000b0290106fab45006so2205420wrl.20
        for <kvm@vger.kernel.org>; Thu, 06 May 2021 06:38:06 -0700 (PDT)
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20161025;
        h=x-gm-message-state:from:to:cc:subject:date:message-id:in-reply-to
         :references:mime-version:content-transfer-encoding;
        bh=/bdOhY3sBOcIHJeoCe6CkJ1QfZvAaT12hvfPxmFulM0=;
        b=RZS87O7GzTn8lL39Rexk2UXU9L3yM0JrQox4X/F47tqa9mGpWAKajgqzoMX/sz75lj
         JjtO1cWZMAcm4BwxIKoiMjFKinDltJv+1I1tDjwhtGWNT7qBgmmGKMUwp0yYAMSqWALy
         6q6KCZcxzFuf9mQIWg0hLfm4azm2a5aJ1yJNWQqpRKahDywXBSfsksD8aqfubr0ZveQw
         j8LgpKNx3pkAfs4NSw9xJE/1e1rCV26B2XumD0+1SJerX/I/FmGy5r8LWtGrkFYrRyr3
         rGIQIxtW21eql/9Pr6AfJirmj+pTiGsRbVdtiQcLghZkhxrjeGCSZemTM27P7yMUw7gg
         OHvA==
X-Gm-Message-State: AOAM531h/HROpZ2bMr1Ci2LfzehQKstnBlsDpJd1wDIE3WU0WqqBSh+e
        j5rywdckh9o3TX6kV7V/3D4/eECEDprhUTILhEd8BGqx3NId+2iYe14Spq8ZGJZtb7QGuTa7x73
        FDyFM5Es+3hcV
X-Received: by 2002:a1c:1bc9:: with SMTP id b192mr15231214wmb.3.1620308285136;
        Thu, 06 May 2021 06:38:05 -0700 (PDT)
X-Google-Smtp-Source: 
 ABdhPJwaU39xWPKmAxzTDnRmk4BpB3yHtP51ZUdmNXUtU3rYDqeE9UGHa8wm/T1yoDDa+r7xsmcCLw==
X-Received: by 2002:a1c:1bc9:: with SMTP id b192mr15231189wmb.3.1620308284991;
        Thu, 06 May 2021 06:38:04 -0700 (PDT)
Received: from localhost.localdomain
 (astrasbourg-652-1-219-60.w90-40.abo.wanadoo.fr. [90.40.114.60])
        by smtp.gmail.com with ESMTPSA id
 o17sm4231200wrs.48.2021.05.06.06.38.04
        (version=TLS1_3 cipher=TLS_AES_256_GCM_SHA384 bits=256/256);
        Thu, 06 May 2021 06:38:04 -0700 (PDT)
From: =?utf-8?q?Philippe_Mathieu-Daud=C3=A9?= <philmd@redhat.com>
To: qemu-devel@nongnu.org
Cc: kvm@vger.kernel.org, qemu-ppc@nongnu.org, qemu-arm@nongnu.org,
 Gerd Hoffmann <kraxel@redhat.com>, Paolo Bonzini <pbonzini@redhat.com>,
	=?utf-8?q?Alex_Benn=C3=A9e?= <alex.bennee@linaro.org>, =?utf-8?q?Philippe_M?=
	=?utf-8?q?athieu-Daud=C3=A9?= <philmd@redhat.com>
Subject: [PATCH v2 1/9] audio/alsaaudio: Replace ALSA alloca() by malloc()
 equivalent
Date: Thu,  6 May 2021 15:37:50 +0200
Message-Id: <20210506133758.1749233-2-philmd@redhat.com>
X-Mailer: git-send-email 2.26.3
In-Reply-To: <20210506133758.1749233-1-philmd@redhat.com>
References: <20210506133758.1749233-1-philmd@redhat.com>
MIME-Version: 1.0
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org

The ALLOCA(3) man-page mentions its "use is discouraged".

Define the cleanup functions for the snd_pcm_[hw/sw]_params_t types,
and replace the ALSA alloca() calls by equivalent ALSA malloc().

Signed-off-by: Philippe Mathieu-Daudé <philmd@redhat.com>
---
 audio/alsaaudio.c | 11 +++++++----
 1 file changed, 7 insertions(+), 4 deletions(-)

diff --git a/audio/alsaaudio.c b/audio/alsaaudio.c
index fcc2f62864f..f39061ebc42 100644
--- a/audio/alsaaudio.c
+++ b/audio/alsaaudio.c
@@ -70,6 +70,9 @@ struct alsa_params_obt {
     snd_pcm_uframes_t samples;
 };
 
+G_DEFINE_AUTOPTR_CLEANUP_FUNC(snd_pcm_hw_params_t, snd_pcm_hw_params_free)
+G_DEFINE_AUTOPTR_CLEANUP_FUNC(snd_pcm_sw_params_t, snd_pcm_sw_params_free)
+
 static void GCC_FMT_ATTR (2, 3) alsa_logerr (int err, const char *fmt, ...)
 {
     va_list ap;
@@ -410,9 +413,9 @@ static void alsa_dump_info (struct alsa_params_req *req,
 static void alsa_set_threshold (snd_pcm_t *handle, snd_pcm_uframes_t threshold)
 {
     int err;
-    snd_pcm_sw_params_t *sw_params;
+    g_autoptr(snd_pcm_sw_params_t) sw_params = NULL;
 
-    snd_pcm_sw_params_alloca (&sw_params);
+    snd_pcm_sw_params_malloc(&sw_params);
 
     err = snd_pcm_sw_params_current (handle, sw_params);
     if (err < 0) {
@@ -444,7 +447,7 @@ static int alsa_open(bool in, struct alsa_params_req *req,
     AudiodevAlsaOptions *aopts = &dev->u.alsa;
     AudiodevAlsaPerDirectionOptions *apdo = in ? aopts->in : aopts->out;
     snd_pcm_t *handle;
-    snd_pcm_hw_params_t *hw_params;
+    g_autoptr(snd_pcm_hw_params_t) hw_params = NULL;
     int err;
     unsigned int freq, nchannels;
     const char *pcm_name = apdo->has_dev ? apdo->dev : "default";
@@ -455,7 +458,7 @@ static int alsa_open(bool in, struct alsa_params_req *req,
     freq = req->freq;
     nchannels = req->nchannels;
 
-    snd_pcm_hw_params_alloca (&hw_params);
+    snd_pcm_hw_params_malloc(&hw_params);
 
     err = snd_pcm_open (
         &handle,

From patchwork Thu May  6 13:37:51 2021
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 8bit
X-Patchwork-Submitter: =?utf-8?q?Philippe_Mathieu-Daud=C3=A9?=
 <philmd@redhat.com>
X-Patchwork-Id: 12242147
Return-Path: <kvm-owner@kernel.org>
X-Spam-Checker-Version: SpamAssassin 3.4.0 (2014-02-07) on
	aws-us-west-2-korg-lkml-1.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No, score=-19.4 required=3.0 tests=BAYES_00,DKIMWL_WL_HIGH,
	DKIM_SIGNED,DKIM_VALID,DKIM_VALID_AU,HEADER_FROM_DIFFERENT_DOMAINS,
	INCLUDES_CR_TRAILER,INCLUDES_PATCH,MAILING_LIST_MULTI,SPF_HELO_NONE,SPF_PASS,
	USER_AGENT_GIT autolearn=ham autolearn_force=no version=3.4.0
Received: from mail.kernel.org (mail.kernel.org [198.145.29.99])
	by smtp.lore.kernel.org (Postfix) with ESMTP id B3B32C433ED
	for <kvm@archiver.kernel.org>; Thu,  6 May 2021 13:38:24 +0000 (UTC)
Received: from vger.kernel.org (vger.kernel.org [23.128.96.18])
	by mail.kernel.org (Postfix) with ESMTP id 80CDB613F3
	for <kvm@archiver.kernel.org>; Thu,  6 May 2021 13:38:24 +0000 (UTC)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S234315AbhEFNjV (ORCPT <rfc822;kvm@archiver.kernel.org>);
        Thu, 6 May 2021 09:39:21 -0400
Received: from us-smtp-delivery-124.mimecast.com ([170.10.133.124]:49980 "EHLO
        us-smtp-delivery-124.mimecast.com" rhost-flags-OK-OK-OK-OK)
        by vger.kernel.org with ESMTP id S234397AbhEFNjP (ORCPT
        <rfc822;kvm@vger.kernel.org>); Thu, 6 May 2021 09:39:15 -0400
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=redhat.com;
        s=mimecast20190719; t=1620308296;
        h=from:from:reply-to:subject:subject:date:date:message-id:message-id:
         to:to:cc:cc:mime-version:mime-version:content-type:content-type:
         content-transfer-encoding:content-transfer-encoding:
         in-reply-to:in-reply-to:references:references;
        bh=Cm48fhhROvZKoiSjxSIRZaypJ/0uOwpu6MiKB+h4o3c=;
        b=QwOA8MGZw5yNjkGiNvOVTG0Lpo2g98zw8x09pI9+On/34si0dFXy1vTya420EDtwUtDbtY
        TTrEJEzW0ASra1HCajfWzJDejZwHuFQe+1QMk64mQ9dQGrwRp+rGdnIaoVdU/szlDviHop
        oXiFPoz7rb0Viz+3T4oFMl4QPfxfHrs=
Received: from mail-wr1-f70.google.com (mail-wr1-f70.google.com
 [209.85.221.70]) (Using TLS) by relay.mimecast.com with ESMTP id
 us-mta-374--U6R7aSvMLqqAmp5pTL9Fw-1; Thu, 06 May 2021 09:38:11 -0400
X-MC-Unique: -U6R7aSvMLqqAmp5pTL9Fw-1
Received: by mail-wr1-f70.google.com with SMTP id
 p19-20020adfc3930000b029010e10128a04so2206105wrf.3
        for <kvm@vger.kernel.org>; Thu, 06 May 2021 06:38:10 -0700 (PDT)
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20161025;
        h=x-gm-message-state:from:to:cc:subject:date:message-id:in-reply-to
         :references:mime-version:content-transfer-encoding;
        bh=Cm48fhhROvZKoiSjxSIRZaypJ/0uOwpu6MiKB+h4o3c=;
        b=E8MbD9gumLgvAswOEXtI23M2TNYWb4PwfbxQ8ONdbtvdjnZsGDing4DRPtYeWcAtta
         QVPN36jxIW7D8oeW612sWuKnrCfQqsy0/tdaTi0mj7I8z2NJL5RxW7x9+ijqJbFJmf2E
         Rt2E84ifHFmoaKkjDIR9Zdpu00h2pYT5Z/+MPHzqV564OCedlA1mr8oqZE8Jr9hNa9za
         4yRU8YLKmzRwelbV6pQtKMx5fqmRW6bANE4aLA/wEajKmPMTFmtKqqcB08qRIfAWk92n
         G6gnBw4y3RW5oMhxY+9PSLAKAoGZ8R3W8OjVqVbhf60eXZ30wq5DJ4SuGk/zL8T8uIbJ
         oaFw==
X-Gm-Message-State: AOAM5327cKg5GzOag836dPNsy6czaDdzLvRKHVz6+NKwOYPCPYIdP5wO
        iMEwYFuqhRTLhuikWYdydN6Qx0lgN7dfprlNZnI3nj+1GjzoNPbylY4euwtF8dJzZFZ1dBYgIjS
        L93ggpWjAORb0
X-Received: by 2002:a5d:64c4:: with SMTP id f4mr5076470wri.178.1620308289927;
        Thu, 06 May 2021 06:38:09 -0700 (PDT)
X-Google-Smtp-Source: 
 ABdhPJy1qLH+TeCACwmkPszo5iNqlWWZLZHj+h1hLu6j1DGNALRbLHKlnd4elCwd5Ol0yGq3nkA99w==
X-Received: by 2002:a5d:64c4:: with SMTP id f4mr5076448wri.178.1620308289821;
        Thu, 06 May 2021 06:38:09 -0700 (PDT)
Received: from localhost.localdomain
 (astrasbourg-652-1-219-60.w90-40.abo.wanadoo.fr. [90.40.114.60])
        by smtp.gmail.com with ESMTPSA id
 j18sm3230757wmq.27.2021.05.06.06.38.08
        (version=TLS1_3 cipher=TLS_AES_256_GCM_SHA384 bits=256/256);
        Thu, 06 May 2021 06:38:09 -0700 (PDT)
From: =?utf-8?q?Philippe_Mathieu-Daud=C3=A9?= <philmd@redhat.com>
To: qemu-devel@nongnu.org
Cc: kvm@vger.kernel.org, qemu-ppc@nongnu.org, qemu-arm@nongnu.org,
 Gerd Hoffmann <kraxel@redhat.com>, Paolo Bonzini <pbonzini@redhat.com>,
	=?utf-8?q?Alex_Benn=C3=A9e?= <alex.bennee@linaro.org>, =?utf-8?q?Philippe_M?=
	=?utf-8?q?athieu-Daud=C3=A9?= <philmd@redhat.com>,
 Stefan Berger <stefanb@linux.vnet.ibm.com>
Subject: [PATCH v2 2/9] backends/tpm: Replace qemu_mutex_lock calls with
 QEMU_LOCK_GUARD
Date: Thu,  6 May 2021 15:37:51 +0200
Message-Id: <20210506133758.1749233-3-philmd@redhat.com>
X-Mailer: git-send-email 2.26.3
In-Reply-To: <20210506133758.1749233-1-philmd@redhat.com>
References: <20210506133758.1749233-1-philmd@redhat.com>
MIME-Version: 1.0
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org

Simplify the tpm_emulator_ctrlcmd() handler by replacing a pair of
qemu_mutex_lock/qemu_mutex_unlock calls by the WITH_QEMU_LOCK_GUARD
macro.

Signed-off-by: Philippe Mathieu-Daudé <philmd@redhat.com>
Reviewed-by: Stefan Berger <stefanb@linux.ibm.com>
Reviewed-by: Christophe de Dinechin <dinechin@redhat.com>
---
 backends/tpm/tpm_emulator.c | 34 +++++++++++++++-------------------
 1 file changed, 15 insertions(+), 19 deletions(-)

diff --git a/backends/tpm/tpm_emulator.c b/backends/tpm/tpm_emulator.c
index a012adc1934..e5f1063ab6c 100644
--- a/backends/tpm/tpm_emulator.c
+++ b/backends/tpm/tpm_emulator.c
@@ -30,6 +30,7 @@
 #include "qemu/error-report.h"
 #include "qemu/module.h"
 #include "qemu/sockets.h"
+#include "qemu/lockable.h"
 #include "io/channel-socket.h"
 #include "sysemu/tpm_backend.h"
 #include "sysemu/tpm_util.h"
@@ -124,31 +125,26 @@ static int tpm_emulator_ctrlcmd(TPMEmulator *tpm, unsigned long cmd, void *msg,
     uint32_t cmd_no = cpu_to_be32(cmd);
     ssize_t n = sizeof(uint32_t) + msg_len_in;
     uint8_t *buf = NULL;
-    int ret = -1;
 
-    qemu_mutex_lock(&tpm->mutex);
+    WITH_QEMU_LOCK_GUARD(&tpm->mutex) {
+        buf = g_alloca(n);
+        memcpy(buf, &cmd_no, sizeof(cmd_no));
+        memcpy(buf + sizeof(cmd_no), msg, msg_len_in);
 
-    buf = g_alloca(n);
-    memcpy(buf, &cmd_no, sizeof(cmd_no));
-    memcpy(buf + sizeof(cmd_no), msg, msg_len_in);
-
-    n = qemu_chr_fe_write_all(dev, buf, n);
-    if (n <= 0) {
-        goto end;
-    }
-
-    if (msg_len_out != 0) {
-        n = qemu_chr_fe_read_all(dev, msg, msg_len_out);
+        n = qemu_chr_fe_write_all(dev, buf, n);
         if (n <= 0) {
-            goto end;
+            return -1;
+        }
+
+        if (msg_len_out != 0) {
+            n = qemu_chr_fe_read_all(dev, msg, msg_len_out);
+            if (n <= 0) {
+                return -1;
+            }
         }
     }
 
-    ret = 0;
-
-end:
-    qemu_mutex_unlock(&tpm->mutex);
-    return ret;
+    return 0;
 }
 
 static int tpm_emulator_unix_tx_bufs(TPMEmulator *tpm_emu,

From patchwork Thu May  6 13:37:52 2021
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 8bit
X-Patchwork-Submitter: =?utf-8?q?Philippe_Mathieu-Daud=C3=A9?=
 <philmd@redhat.com>
X-Patchwork-Id: 12242149
Return-Path: <kvm-owner@kernel.org>
X-Spam-Checker-Version: SpamAssassin 3.4.0 (2014-02-07) on
	aws-us-west-2-korg-lkml-1.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No, score=-19.4 required=3.0 tests=BAYES_00,DKIMWL_WL_HIGH,
	DKIM_SIGNED,DKIM_VALID,DKIM_VALID_AU,HEADER_FROM_DIFFERENT_DOMAINS,
	INCLUDES_CR_TRAILER,INCLUDES_PATCH,MAILING_LIST_MULTI,SPF_HELO_NONE,SPF_PASS,
	USER_AGENT_GIT autolearn=ham autolearn_force=no version=3.4.0
Received: from mail.kernel.org (mail.kernel.org [198.145.29.99])
	by smtp.lore.kernel.org (Postfix) with ESMTP id 1031CC433B4
	for <kvm@archiver.kernel.org>; Thu,  6 May 2021 13:38:26 +0000 (UTC)
Received: from vger.kernel.org (vger.kernel.org [23.128.96.18])
	by mail.kernel.org (Postfix) with ESMTP id E11A2613F1
	for <kvm@archiver.kernel.org>; Thu,  6 May 2021 13:38:25 +0000 (UTC)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S234300AbhEFNjW (ORCPT <rfc822;kvm@archiver.kernel.org>);
        Thu, 6 May 2021 09:39:22 -0400
Received: from us-smtp-delivery-124.mimecast.com ([216.205.24.124]:48068 "EHLO
        us-smtp-delivery-124.mimecast.com" rhost-flags-OK-OK-OK-OK)
        by vger.kernel.org with ESMTP id S234466AbhEFNjT (ORCPT
        <rfc822;kvm@vger.kernel.org>); Thu, 6 May 2021 09:39:19 -0400
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=redhat.com;
        s=mimecast20190719; t=1620308301;
        h=from:from:reply-to:subject:subject:date:date:message-id:message-id:
         to:to:cc:cc:mime-version:mime-version:content-type:content-type:
         content-transfer-encoding:content-transfer-encoding:
         in-reply-to:in-reply-to:references:references;
        bh=rtnD1U4zAjjBX9oA9uBgg0xbiA9djz1nnWZVPwnN9No=;
        b=E/hX3FD+jj21UDon5hGG5vq5140HF3H9GHxZhMjjyYFxDJPdvhtuT239XCREI3SqisyA2p
        z63rMCDYe0rU5q7eKgCfjvmCPgtmmLfq0E2c4P3MGDffoNFxhNq6rh9jni4Ab2uDlLTKVV
        eiBBrzGC/GE23M8kAWc5RNJjuCD/nsU=
Received: from mail-wm1-f71.google.com (mail-wm1-f71.google.com
 [209.85.128.71]) (Using TLS) by relay.mimecast.com with ESMTP id
 us-mta-507-KAa5CyKqO2aQziOW-RYybA-1; Thu, 06 May 2021 09:38:16 -0400
X-MC-Unique: KAa5CyKqO2aQziOW-RYybA-1
Received: by mail-wm1-f71.google.com with SMTP id
 n24-20020a7bcbd80000b029014287841063so1346518wmi.3
        for <kvm@vger.kernel.org>; Thu, 06 May 2021 06:38:16 -0700 (PDT)
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20161025;
        h=x-gm-message-state:from:to:cc:subject:date:message-id:in-reply-to
         :references:mime-version:content-transfer-encoding;
        bh=rtnD1U4zAjjBX9oA9uBgg0xbiA9djz1nnWZVPwnN9No=;
        b=DXTXGoAsLgZvJTvGIiDf1oy7C1ipW4NtxFEcV6ulin0FnqG8jNAbz2uvukw1Z6Co9m
         beAA2MMg81rhmH1ejtvzD967VhpKg7xaiHtn5/warYdnE2xoPCONLlGapprz7fs0t3Ap
         m2g7JTxSP2w5v2bBLYIknWSwTtxBEKmYOQl+9tVSw+iqr/fdcVrUvC1ym5XAzBKDO890
         gKFxXm7GP7B9V8l7ZNdpphuv8UnSnJRnnPSXEqMqPoj4a3c01U3AYfC6pzOweK150ZgT
         jk/M8YTM538OVjs1eYpBHU+wEp1Gw+Z9wqkIA+zS9IduGKMzz+thnEyBVmzpBuS4O+u9
         00wQ==
X-Gm-Message-State: AOAM530AHyBDsk3RuuqR10V4H79ICQSwNnOGNw1OB8fyQjF/wR5gizwR
        fyIrtdjdJOzQ2VATgFILc9y+sTjFvKhr8kbstuALoPX+eIKsobbyYN8b7kqNyMWx/Rj7WjmR9nI
        2sY2rchXrJWUr
X-Received: by 2002:a1c:c28a:: with SMTP id
 s132mr15033742wmf.145.1620308295092;
        Thu, 06 May 2021 06:38:15 -0700 (PDT)
X-Google-Smtp-Source: 
 ABdhPJx9zr0u80TiOLaMnzP6Zw8/LeB0azlBgPsu0O1udkYDmO7GAIcBjFqIX7hcrVdU+Gtn8cUBtA==
X-Received: by 2002:a1c:c28a:: with SMTP id
 s132mr15033720wmf.145.1620308294899;
        Thu, 06 May 2021 06:38:14 -0700 (PDT)
Received: from localhost.localdomain
 (astrasbourg-652-1-219-60.w90-40.abo.wanadoo.fr. [90.40.114.60])
        by smtp.gmail.com with ESMTPSA id
 o15sm4312755wru.42.2021.05.06.06.38.13
        (version=TLS1_3 cipher=TLS_AES_256_GCM_SHA384 bits=256/256);
        Thu, 06 May 2021 06:38:14 -0700 (PDT)
From: =?utf-8?q?Philippe_Mathieu-Daud=C3=A9?= <philmd@redhat.com>
To: qemu-devel@nongnu.org
Cc: kvm@vger.kernel.org, qemu-ppc@nongnu.org, qemu-arm@nongnu.org,
 Gerd Hoffmann <kraxel@redhat.com>, Paolo Bonzini <pbonzini@redhat.com>,
	=?utf-8?q?Alex_Benn=C3=A9e?= <alex.bennee@linaro.org>, =?utf-8?q?Philippe_M?=
	=?utf-8?q?athieu-Daud=C3=A9?= <philmd@redhat.com>,
 Stefan Berger <stefanb@linux.vnet.ibm.com>
Subject: [PATCH v2 3/9] backends/tpm: Replace g_alloca() by g_malloc()
Date: Thu,  6 May 2021 15:37:52 +0200
Message-Id: <20210506133758.1749233-4-philmd@redhat.com>
X-Mailer: git-send-email 2.26.3
In-Reply-To: <20210506133758.1749233-1-philmd@redhat.com>
References: <20210506133758.1749233-1-philmd@redhat.com>
MIME-Version: 1.0
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org

The ALLOCA(3) man-page mentions its "use is discouraged".

Replace a g_alloca() call by a g_malloc() one, moving the
allocation before the MUTEX guarded block.

Signed-off-by: Philippe Mathieu-Daudé <philmd@redhat.com>
Reviewed-by: Stefan Berger <stefanb@linux.ibm.com>
---
 backends/tpm/tpm_emulator.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/backends/tpm/tpm_emulator.c b/backends/tpm/tpm_emulator.c
index e5f1063ab6c..d37a6d563a3 100644
--- a/backends/tpm/tpm_emulator.c
+++ b/backends/tpm/tpm_emulator.c
@@ -124,10 +124,9 @@ static int tpm_emulator_ctrlcmd(TPMEmulator *tpm, unsigned long cmd, void *msg,
     CharBackend *dev = &tpm->ctrl_chr;
     uint32_t cmd_no = cpu_to_be32(cmd);
     ssize_t n = sizeof(uint32_t) + msg_len_in;
-    uint8_t *buf = NULL;
+    g_autofree uint8_t *buf = g_malloc(n);
 
     WITH_QEMU_LOCK_GUARD(&tpm->mutex) {
-        buf = g_alloca(n);
         memcpy(buf, &cmd_no, sizeof(cmd_no));
         memcpy(buf + sizeof(cmd_no), msg, msg_len_in);
 

From patchwork Thu May  6 13:37:53 2021
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 8bit
X-Patchwork-Submitter: =?utf-8?q?Philippe_Mathieu-Daud=C3=A9?=
 <philmd@redhat.com>
X-Patchwork-Id: 12242151
Return-Path: <kvm-owner@kernel.org>
X-Spam-Checker-Version: SpamAssassin 3.4.0 (2014-02-07) on
	aws-us-west-2-korg-lkml-1.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No, score=-19.4 required=3.0 tests=BAYES_00,DKIMWL_WL_HIGH,
	DKIM_SIGNED,DKIM_VALID,DKIM_VALID_AU,HEADER_FROM_DIFFERENT_DOMAINS,
	INCLUDES_CR_TRAILER,INCLUDES_PATCH,MAILING_LIST_MULTI,SPF_HELO_NONE,SPF_PASS,
	USER_AGENT_GIT autolearn=ham autolearn_force=no version=3.4.0
Received: from mail.kernel.org (mail.kernel.org [198.145.29.99])
	by smtp.lore.kernel.org (Postfix) with ESMTP id F3982C433ED
	for <kvm@archiver.kernel.org>; Thu,  6 May 2021 13:38:29 +0000 (UTC)
Received: from vger.kernel.org (vger.kernel.org [23.128.96.18])
	by mail.kernel.org (Postfix) with ESMTP id C43B7613F3
	for <kvm@archiver.kernel.org>; Thu,  6 May 2021 13:38:29 +0000 (UTC)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S234072AbhEFNj1 (ORCPT <rfc822;kvm@archiver.kernel.org>);
        Thu, 6 May 2021 09:39:27 -0400
Received: from us-smtp-delivery-124.mimecast.com ([216.205.24.124]:28872 "EHLO
        us-smtp-delivery-124.mimecast.com" rhost-flags-OK-OK-OK-OK)
        by vger.kernel.org with ESMTP id S234348AbhEFNjZ (ORCPT
        <rfc822;kvm@vger.kernel.org>); Thu, 6 May 2021 09:39:25 -0400
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=redhat.com;
        s=mimecast20190719; t=1620308307;
        h=from:from:reply-to:subject:subject:date:date:message-id:message-id:
         to:to:cc:cc:mime-version:mime-version:content-type:content-type:
         content-transfer-encoding:content-transfer-encoding:
         in-reply-to:in-reply-to:references:references;
        bh=dXiqUMGNseF8aEoCidrrmNydk4TNmOpI8OAZgIN0M9M=;
        b=ZjK+yq2FgDa3M9koIy9tQP+h0mcn3lSxVNWVW+IfS4QcOux4DH/QOhXl/K3TQThln0d1ba
        42snp0tIgz+3w6qgeOnLDFB97e86loPbhY5VOcbpjwDW8FdsUrW/37M5s84aPIZSUpxc0i
        XrJpQsy5un10T9hDoXX38nzhDe+htnI=
Received: from mail-wr1-f69.google.com (mail-wr1-f69.google.com
 [209.85.221.69]) (Using TLS) by relay.mimecast.com with ESMTP id
 us-mta-567-4KCL0lJFM6Cwcf2iTjaPpA-1; Thu, 06 May 2021 09:38:21 -0400
X-MC-Unique: 4KCL0lJFM6Cwcf2iTjaPpA-1
Received: by mail-wr1-f69.google.com with SMTP id
 r12-20020adfc10c0000b029010d83323601so2191695wre.22
        for <kvm@vger.kernel.org>; Thu, 06 May 2021 06:38:20 -0700 (PDT)
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20161025;
        h=x-gm-message-state:from:to:cc:subject:date:message-id:in-reply-to
         :references:mime-version:content-transfer-encoding;
        bh=dXiqUMGNseF8aEoCidrrmNydk4TNmOpI8OAZgIN0M9M=;
        b=tfihd//Ze6GVF4RtchgB/C9CDX+25f+hZLb7fhvZ9LAVQg4M5nyDyPYqB2HhKGfhUR
         wSrc0ycYiFApyVYJw4vBFBMA/gj5kTXrUJwNrsBqSo4U2wCgTSRyoYti/jST+xrrQ4QU
         dMdQgb5Js47qYJqHhtanOnSWt1uupPyc8Lr/KuWzRjsBlQhg+LR7bryPnTNnHoXN8RC6
         YEnsdszzyFAJ8bWsp7Z7/LPGyEBeuhPGxMvVSH2Qw03OU8fo/YoITfhRWyc5Yy6NROY1
         OnQ8nNGI6rqK8BM922l2VU140ZXU9qPQdcCReQ38Z2ZL+cmu0MlMZInIQkrwZU09/jWk
         Fn7Q==
X-Gm-Message-State: AOAM532mIt09+Ms/pq1HC34Q9+A+fod2j/rS6f2BuJIsKiAC0h/0Mx1n
        Mp6i9NUBegM6DdVuapaGMdmN6hvW0KQlEI0Y7TV4RDpPNBTOilslEHX5+P0YlsYEZvGyJX+cp2F
        1r+CWInxymkRM
X-Received: by 2002:a5d:51ce:: with SMTP id n14mr5147672wrv.239.1620308299896;
        Thu, 06 May 2021 06:38:19 -0700 (PDT)
X-Google-Smtp-Source: 
 ABdhPJzyAZIfvg++T6fTYKrEquhu6tnCC62wCEQ30o8S5aLEgqSCKFWctM9XzEDZQWk7c63Etg+U8g==
X-Received: by 2002:a5d:51ce:: with SMTP id n14mr5147655wrv.239.1620308299780;
        Thu, 06 May 2021 06:38:19 -0700 (PDT)
Received: from localhost.localdomain
 (astrasbourg-652-1-219-60.w90-40.abo.wanadoo.fr. [90.40.114.60])
        by smtp.gmail.com with ESMTPSA id
 m13sm4432533wrw.86.2021.05.06.06.38.18
        (version=TLS1_3 cipher=TLS_AES_256_GCM_SHA384 bits=256/256);
        Thu, 06 May 2021 06:38:19 -0700 (PDT)
From: =?utf-8?q?Philippe_Mathieu-Daud=C3=A9?= <philmd@redhat.com>
To: qemu-devel@nongnu.org
Cc: kvm@vger.kernel.org, qemu-ppc@nongnu.org, qemu-arm@nongnu.org,
 Gerd Hoffmann <kraxel@redhat.com>, Paolo Bonzini <pbonzini@redhat.com>,
	=?utf-8?q?Alex_Benn=C3=A9e?= <alex.bennee@linaro.org>, =?utf-8?q?Philippe_M?=
	=?utf-8?q?athieu-Daud=C3=A9?= <philmd@redhat.com>,
 Warner Losh <imp@bsdimp.com>, Kyle Evans <kevans@freebsd.org>
Subject: [PATCH v2 4/9] bsd-user/syscall: Replace alloca() by g_new()
Date: Thu,  6 May 2021 15:37:53 +0200
Message-Id: <20210506133758.1749233-5-philmd@redhat.com>
X-Mailer: git-send-email 2.26.3
In-Reply-To: <20210506133758.1749233-1-philmd@redhat.com>
References: <20210506133758.1749233-1-philmd@redhat.com>
MIME-Version: 1.0
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org

The ALLOCA(3) man-page mentions its "use is discouraged".

Replace it by a g_new() call.

Signed-off-by: Philippe Mathieu-Daudé <philmd@redhat.com>
---
 bsd-user/syscall.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/bsd-user/syscall.c b/bsd-user/syscall.c
index 4abff796c76..dbee0385ceb 100644
--- a/bsd-user/syscall.c
+++ b/bsd-user/syscall.c
@@ -355,9 +355,8 @@ abi_long do_freebsd_syscall(void *cpu_env, int num, abi_long arg1,
     case TARGET_FREEBSD_NR_writev:
         {
             int count = arg3;
-            struct iovec *vec;
+            g_autofree struct iovec *vec = g_new(struct iovec, count);
 
-            vec = alloca(count * sizeof(struct iovec));
             if (lock_iovec(VERIFY_READ, vec, arg2, count, 1) < 0)
                 goto efault;
             ret = get_errno(writev(arg1, vec, count));

From patchwork Thu May  6 13:37:54 2021
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 8bit
X-Patchwork-Submitter: =?utf-8?q?Philippe_Mathieu-Daud=C3=A9?=
 <philmd@redhat.com>
X-Patchwork-Id: 12242153
Return-Path: <kvm-owner@kernel.org>
X-Spam-Checker-Version: SpamAssassin 3.4.0 (2014-02-07) on
	aws-us-west-2-korg-lkml-1.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No, score=-19.4 required=3.0 tests=BAYES_00,DKIMWL_WL_HIGH,
	DKIM_SIGNED,DKIM_VALID,DKIM_VALID_AU,HEADER_FROM_DIFFERENT_DOMAINS,
	INCLUDES_CR_TRAILER,INCLUDES_PATCH,MAILING_LIST_MULTI,SPF_HELO_NONE,SPF_PASS,
	USER_AGENT_GIT autolearn=ham autolearn_force=no version=3.4.0
Received: from mail.kernel.org (mail.kernel.org [198.145.29.99])
	by smtp.lore.kernel.org (Postfix) with ESMTP id AE6ADC433B4
	for <kvm@archiver.kernel.org>; Thu,  6 May 2021 13:38:30 +0000 (UTC)
Received: from vger.kernel.org (vger.kernel.org [23.128.96.18])
	by mail.kernel.org (Postfix) with ESMTP id 7400D613F3
	for <kvm@archiver.kernel.org>; Thu,  6 May 2021 13:38:30 +0000 (UTC)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S234340AbhEFNj1 (ORCPT <rfc822;kvm@archiver.kernel.org>);
        Thu, 6 May 2021 09:39:27 -0400
Received: from us-smtp-delivery-124.mimecast.com ([216.205.24.124]:25699 "EHLO
        us-smtp-delivery-124.mimecast.com" rhost-flags-OK-OK-OK-OK)
        by vger.kernel.org with ESMTP id S234282AbhEFNjZ (ORCPT
        <rfc822;kvm@vger.kernel.org>); Thu, 6 May 2021 09:39:25 -0400
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=redhat.com;
        s=mimecast20190719; t=1620308307;
        h=from:from:reply-to:subject:subject:date:date:message-id:message-id:
         to:to:cc:cc:mime-version:mime-version:content-type:content-type:
         content-transfer-encoding:content-transfer-encoding:
         in-reply-to:in-reply-to:references:references;
        bh=B3vDcPc2Lwo5z0s8ItWutJUQDssHj4SCpM1PcCRAn5U=;
        b=QRau2uHS6G+HWVTwmcDOgVnRuVh3MkMaP2cYoeOnezZG8xw6K5TwSN4n4uLMsAtuWe7x6y
        OVbGWnk3NbBR+1aompcBchjeJ8nAtI25UpR/fYlEPHXwIhxad7WuoUfIf5Vd77yV+gJce6
        CkEnqrYkWeCNuMJBzkDSftPQ5JazNS8=
Received: from mail-wm1-f69.google.com (mail-wm1-f69.google.com
 [209.85.128.69]) (Using TLS) by relay.mimecast.com with ESMTP id
 us-mta-23-yBaaPpAWOvGutJNKz-xXpA-1; Thu, 06 May 2021 09:38:25 -0400
X-MC-Unique: yBaaPpAWOvGutJNKz-xXpA-1
Received: by mail-wm1-f69.google.com with SMTP id
 d199-20020a1c1dd00000b02901492c14476eso1345301wmd.2
        for <kvm@vger.kernel.org>; Thu, 06 May 2021 06:38:25 -0700 (PDT)
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20161025;
        h=x-gm-message-state:from:to:cc:subject:date:message-id:in-reply-to
         :references:mime-version:content-transfer-encoding;
        bh=B3vDcPc2Lwo5z0s8ItWutJUQDssHj4SCpM1PcCRAn5U=;
        b=ll7OBTLWh64obsrrqkS80P0wOazTKUuADTcPtElrSvIy3+dN/cjdBv8He8snX62IER
         sNi3ff944q/MY4ZeMfuFR0rstYfVsMar1bvvF8dhhYrYCao5JFr47FLyXAhIGwxSIAQQ
         zRFyR9P/opFxg/OwsdbwepE+7XCAaFQ5yCdHOmBkD1GS3ITz5Urkcb0AtD+FEw5R7dJp
         h9byFej1KBu55XJOjv6qDpbzJGJbRSdxKKo/PR/4qyCDSsv648m0L9Ia3VX2YRr+cwH8
         r8Mwxp/6N5iEAqsoRM58gyf0vf03jyW7TmdOmK1wWyBAseOBEeGWcaqRyjphsNoELCZe
         o/9g==
X-Gm-Message-State: AOAM530X8z1qRyS4BCwI94QfKadO929c7+tT/GpkZQ0bOOICLFzgQQ15
        ap6bKV5F8hwj5JmI6/KVoaytwJSmNsv3nhAKJnwp89RzbtbbC3AWJ5JxAQvb0rP/vVY+V6oSbYe
        LXIaUu2czvCDh
X-Received: by 2002:a5d:48c3:: with SMTP id p3mr4659223wrs.150.1620308304640;
        Thu, 06 May 2021 06:38:24 -0700 (PDT)
X-Google-Smtp-Source: 
 ABdhPJwECKu/BlFqjZgy5iMIlrOtoOiBjrMgG2/qWWwpALxiW11/+mDQsXIqidjTaJhIgu3oXy9nPg==
X-Received: by 2002:a5d:48c3:: with SMTP id p3mr4659202wrs.150.1620308304482;
        Thu, 06 May 2021 06:38:24 -0700 (PDT)
Received: from localhost.localdomain
 (astrasbourg-652-1-219-60.w90-40.abo.wanadoo.fr. [90.40.114.60])
        by smtp.gmail.com with ESMTPSA id
 c15sm4424312wrr.3.2021.05.06.06.38.23
        (version=TLS1_3 cipher=TLS_AES_256_GCM_SHA384 bits=256/256);
        Thu, 06 May 2021 06:38:24 -0700 (PDT)
From: =?utf-8?q?Philippe_Mathieu-Daud=C3=A9?= <philmd@redhat.com>
To: qemu-devel@nongnu.org
Cc: kvm@vger.kernel.org, qemu-ppc@nongnu.org, qemu-arm@nongnu.org,
 Gerd Hoffmann <kraxel@redhat.com>, Paolo Bonzini <pbonzini@redhat.com>,
	=?utf-8?q?Alex_Benn=C3=A9e?= <alex.bennee@linaro.org>, =?utf-8?q?Philippe_M?=
	=?utf-8?q?athieu-Daud=C3=A9?= <philmd@redhat.com>
Subject: [PATCH v2 5/9] gdbstub: Constify GdbCmdParseEntry
Date: Thu,  6 May 2021 15:37:54 +0200
Message-Id: <20210506133758.1749233-6-philmd@redhat.com>
X-Mailer: git-send-email 2.26.3
In-Reply-To: <20210506133758.1749233-1-philmd@redhat.com>
References: <20210506133758.1749233-1-philmd@redhat.com>
MIME-Version: 1.0
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org

Reviewed-by: Alex Bennée <alex.bennee@linaro.org>
Signed-off-by: Philippe Mathieu-Daudé <philmd@redhat.com>
---
 gdbstub.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/gdbstub.c b/gdbstub.c
index 9103ffc9028..83d47c67325 100644
--- a/gdbstub.c
+++ b/gdbstub.c
@@ -1981,7 +1981,7 @@ static void handle_v_kill(GdbCmdContext *gdb_ctx, void *user_ctx)
     exit(0);
 }
 
-static GdbCmdParseEntry gdb_v_commands_table[] = {
+static const GdbCmdParseEntry gdb_v_commands_table[] = {
     /* Order is important if has same prefix */
     {
         .handler = handle_v_cont_query,
@@ -2324,7 +2324,7 @@ static void handle_set_qemu_phy_mem_mode(GdbCmdContext *gdb_ctx, void *user_ctx)
 }
 #endif
 
-static GdbCmdParseEntry gdb_gen_query_set_common_table[] = {
+static const GdbCmdParseEntry gdb_gen_query_set_common_table[] = {
     /* Order is important if has same prefix */
     {
         .handler = handle_query_qemu_sstepbits,
@@ -2342,7 +2342,7 @@ static GdbCmdParseEntry gdb_gen_query_set_common_table[] = {
     },
 };
 
-static GdbCmdParseEntry gdb_gen_query_table[] = {
+static const GdbCmdParseEntry gdb_gen_query_table[] = {
     {
         .handler = handle_query_curr_tid,
         .cmd = "C",
@@ -2420,7 +2420,7 @@ static GdbCmdParseEntry gdb_gen_query_table[] = {
 #endif
 };
 
-static GdbCmdParseEntry gdb_gen_set_table[] = {
+static const GdbCmdParseEntry gdb_gen_set_table[] = {
     /* Order is important if has same prefix */
     {
         .handler = handle_set_qemu_sstep,

From patchwork Thu May  6 13:37:55 2021
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 8bit
X-Patchwork-Submitter: =?utf-8?q?Philippe_Mathieu-Daud=C3=A9?=
 <philmd@redhat.com>
X-Patchwork-Id: 12242155
Return-Path: <kvm-owner@kernel.org>
X-Spam-Checker-Version: SpamAssassin 3.4.0 (2014-02-07) on
	aws-us-west-2-korg-lkml-1.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No, score=-19.4 required=3.0 tests=BAYES_00,DKIMWL_WL_HIGH,
	DKIM_SIGNED,DKIM_VALID,DKIM_VALID_AU,HEADER_FROM_DIFFERENT_DOMAINS,
	INCLUDES_CR_TRAILER,INCLUDES_PATCH,MAILING_LIST_MULTI,SPF_HELO_NONE,SPF_PASS,
	USER_AGENT_GIT autolearn=ham autolearn_force=no version=3.4.0
Received: from mail.kernel.org (mail.kernel.org [198.145.29.99])
	by smtp.lore.kernel.org (Postfix) with ESMTP id 8C433C433B4
	for <kvm@archiver.kernel.org>; Thu,  6 May 2021 13:38:34 +0000 (UTC)
Received: from vger.kernel.org (vger.kernel.org [23.128.96.18])
	by mail.kernel.org (Postfix) with ESMTP id 58A53613F3
	for <kvm@archiver.kernel.org>; Thu,  6 May 2021 13:38:34 +0000 (UTC)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S234282AbhEFNjb (ORCPT <rfc822;kvm@archiver.kernel.org>);
        Thu, 6 May 2021 09:39:31 -0400
Received: from us-smtp-delivery-124.mimecast.com ([170.10.133.124]:36519 "EHLO
        us-smtp-delivery-124.mimecast.com" rhost-flags-OK-OK-OK-OK)
        by vger.kernel.org with ESMTP id S233954AbhEFNja (ORCPT
        <rfc822;kvm@vger.kernel.org>); Thu, 6 May 2021 09:39:30 -0400
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=redhat.com;
        s=mimecast20190719; t=1620308311;
        h=from:from:reply-to:subject:subject:date:date:message-id:message-id:
         to:to:cc:cc:mime-version:mime-version:content-type:content-type:
         content-transfer-encoding:content-transfer-encoding:
         in-reply-to:in-reply-to:references:references;
        bh=dmSw65R373KTKOSFnzTrI/V4leURDKdprV/OcTyehJU=;
        b=DUYvbRdOTNnvbwjouOuuEQ6NUL0/7wOJIn5PcZ6J4moOrAmK3itbU1lhvV++kbf0HmQCM8
        9hVaoBqMpbqMP6eZyYyjL8JOvj6a0A57c0dOPvo6mzg+QdCrpMMICf8/oTUn79aY9RWnSO
        o3mN/1Cb8nUXw/Oz+0HvXyk+dalshUA=
Received: from mail-wr1-f72.google.com (mail-wr1-f72.google.com
 [209.85.221.72]) (Using TLS) by relay.mimecast.com with ESMTP id
 us-mta-560-4WEe88tvOiaHNf_OGb5Khg-1; Thu, 06 May 2021 09:38:30 -0400
X-MC-Unique: 4WEe88tvOiaHNf_OGb5Khg-1
Received: by mail-wr1-f72.google.com with SMTP id
 4-20020adf80040000b029010cab735fdeso2206938wrk.14
        for <kvm@vger.kernel.org>; Thu, 06 May 2021 06:38:30 -0700 (PDT)
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20161025;
        h=x-gm-message-state:from:to:cc:subject:date:message-id:in-reply-to
         :references:mime-version:content-transfer-encoding;
        bh=dmSw65R373KTKOSFnzTrI/V4leURDKdprV/OcTyehJU=;
        b=TIAx9I/lclYacgqowSOE2GSsZXOrEiFU4dfRPE3gTFNb9dIa/G5Zi+9lNVZixkV5/Y
         xV16ZDEn5ZCh6QXyPNKQvTcjwKXe61QEJB8tSfQAA/i234ieClXbLJFMBtLUNazylMIt
         jPUOgyu+PSFr6rLKFhae8FtZzCx+wuevRw36gb9qJEzq6EPN5pJ4Mdn9LxnbZFMtcWgs
         H7m000kMsQBQMypJU8lepn9C/+FIVcVd9tA0AIKSWZazIq1hU5nu2/+iuM7PGlHdpDDe
         F3W62gZ1WKUtq6ouPnN7QnxN8NEVfn8Ym0KCzAf4K1Ldi9TKBtt0q6fEnq515I+F06ho
         3fQQ==
X-Gm-Message-State: AOAM530LFDNx7OkOXl+1bsdO1T9gXo5KSmKODV6XPbzfuozATXX18BVf
        55Phxwv/UOz60Qg87Xm2LIeiz6E4x00odPRG41m5x+ZSxIv9BWV0DbSvSVlkylkmZV9zoiUfPP1
        UYkHSczUpBqZv
X-Received: by 2002:a05:600c:4f0f:: with SMTP id
 l15mr15225987wmq.143.1620308309316;
        Thu, 06 May 2021 06:38:29 -0700 (PDT)
X-Google-Smtp-Source: 
 ABdhPJzDIDMczepsuakeS2rOwouy+q2KMwVcj0QDOZPZg65zS1H29vCOvRHBjDW4Th8rjGTb2H7EvA==
X-Received: by 2002:a05:600c:4f0f:: with SMTP id
 l15mr15225973wmq.143.1620308309193;
        Thu, 06 May 2021 06:38:29 -0700 (PDT)
Received: from localhost.localdomain
 (astrasbourg-652-1-219-60.w90-40.abo.wanadoo.fr. [90.40.114.60])
        by smtp.gmail.com with ESMTPSA id
 d3sm4089244wri.75.2021.05.06.06.38.28
        (version=TLS1_3 cipher=TLS_AES_256_GCM_SHA384 bits=256/256);
        Thu, 06 May 2021 06:38:28 -0700 (PDT)
From: =?utf-8?q?Philippe_Mathieu-Daud=C3=A9?= <philmd@redhat.com>
To: qemu-devel@nongnu.org
Cc: kvm@vger.kernel.org, qemu-ppc@nongnu.org, qemu-arm@nongnu.org,
 Gerd Hoffmann <kraxel@redhat.com>, Paolo Bonzini <pbonzini@redhat.com>,
	=?utf-8?q?Alex_Benn=C3=A9e?= <alex.bennee@linaro.org>, =?utf-8?q?Philippe_M?=
	=?utf-8?q?athieu-Daud=C3=A9?= <philmd@redhat.com>
Subject: [PATCH v2 6/9] gdbstub: Only call cmd_parse_params() with non-NULL
 command schema
Date: Thu,  6 May 2021 15:37:55 +0200
Message-Id: <20210506133758.1749233-7-philmd@redhat.com>
X-Mailer: git-send-email 2.26.3
In-Reply-To: <20210506133758.1749233-1-philmd@redhat.com>
References: <20210506133758.1749233-1-philmd@redhat.com>
MIME-Version: 1.0
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org

Move the NULL check on command schema buffer from the callee
cmd_parse_params() to the single caller, process_string_cmd().

This simplifies the process_string_cmd() logic.

Signed-off-by: Philippe Mathieu-Daudé <philmd@redhat.com>
Reviewed-by: Alex Bennée <alex.bennee@linaro.org>
---
 gdbstub.c | 27 ++++++++++++---------------
 1 file changed, 12 insertions(+), 15 deletions(-)

diff --git a/gdbstub.c b/gdbstub.c
index 83d47c67325..7cee2fb0f1f 100644
--- a/gdbstub.c
+++ b/gdbstub.c
@@ -1368,12 +1368,9 @@ static int cmd_parse_params(const char *data, const char *schema,
     int curr_param;
     const char *curr_schema, *curr_data;
 
+    assert(schema);
     *num_params = 0;
 
-    if (!schema) {
-        return 0;
-    }
-
     curr_schema = schema;
     curr_param = 0;
     curr_data = data;
@@ -1471,7 +1468,7 @@ static inline int startswith(const char *string, const char *pattern)
 static int process_string_cmd(void *user_ctx, const char *data,
                               const GdbCmdParseEntry *cmds, int num_cmds)
 {
-    int i, schema_len, max_num_params = 0;
+    int i;
     GdbCmdContext gdb_ctx;
 
     if (!cmds) {
@@ -1488,21 +1485,21 @@ static int process_string_cmd(void *user_ctx, const char *data,
         }
 
         if (cmd->schema) {
-            schema_len = strlen(cmd->schema);
+            int schema_len = strlen(cmd->schema);
+            int max_num_params = schema_len / 2;
+
             if (schema_len % 2) {
                 return -2;
             }
 
-            max_num_params = schema_len / 2;
-        }
+            gdb_ctx.params = (GdbCmdVariant *)alloca(sizeof(*gdb_ctx.params)
+                                                     * max_num_params);
+            memset(gdb_ctx.params, 0, sizeof(*gdb_ctx.params) * max_num_params);
 
-        gdb_ctx.params =
-            (GdbCmdVariant *)alloca(sizeof(*gdb_ctx.params) * max_num_params);
-        memset(gdb_ctx.params, 0, sizeof(*gdb_ctx.params) * max_num_params);
-
-        if (cmd_parse_params(&data[strlen(cmd->cmd)], cmd->schema,
-                             gdb_ctx.params, &gdb_ctx.num_params)) {
-            return -1;
+            if (cmd_parse_params(&data[strlen(cmd->cmd)], cmd->schema,
+                                 gdb_ctx.params, &gdb_ctx.num_params)) {
+                return -1;
+            }
         }
 
         cmd->handler(&gdb_ctx, user_ctx);

From patchwork Thu May  6 13:37:56 2021
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 8bit
X-Patchwork-Submitter: =?utf-8?q?Philippe_Mathieu-Daud=C3=A9?=
 <philmd@redhat.com>
X-Patchwork-Id: 12242157
Return-Path: <kvm-owner@kernel.org>
X-Spam-Checker-Version: SpamAssassin 3.4.0 (2014-02-07) on
	aws-us-west-2-korg-lkml-1.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No, score=-19.4 required=3.0 tests=BAYES_00,DKIMWL_WL_HIGH,
	DKIM_SIGNED,DKIM_VALID,DKIM_VALID_AU,HEADER_FROM_DIFFERENT_DOMAINS,
	INCLUDES_CR_TRAILER,INCLUDES_PATCH,MAILING_LIST_MULTI,SPF_HELO_NONE,SPF_PASS,
	USER_AGENT_GIT autolearn=ham autolearn_force=no version=3.4.0
Received: from mail.kernel.org (mail.kernel.org [198.145.29.99])
	by smtp.lore.kernel.org (Postfix) with ESMTP id 3D687C433ED
	for <kvm@archiver.kernel.org>; Thu,  6 May 2021 13:38:39 +0000 (UTC)
Received: from vger.kernel.org (vger.kernel.org [23.128.96.18])
	by mail.kernel.org (Postfix) with ESMTP id 1AB29613F3
	for <kvm@archiver.kernel.org>; Thu,  6 May 2021 13:38:39 +0000 (UTC)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S234357AbhEFNjg (ORCPT <rfc822;kvm@archiver.kernel.org>);
        Thu, 6 May 2021 09:39:36 -0400
Received: from us-smtp-delivery-124.mimecast.com ([216.205.24.124]:45187 "EHLO
        us-smtp-delivery-124.mimecast.com" rhost-flags-OK-OK-OK-OK)
        by vger.kernel.org with ESMTP id S233954AbhEFNjf (ORCPT
        <rfc822;kvm@vger.kernel.org>); Thu, 6 May 2021 09:39:35 -0400
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=redhat.com;
        s=mimecast20190719; t=1620308316;
        h=from:from:reply-to:subject:subject:date:date:message-id:message-id:
         to:to:cc:cc:mime-version:mime-version:content-type:content-type:
         content-transfer-encoding:content-transfer-encoding:
         in-reply-to:in-reply-to:references:references;
        bh=k4PLS6ZKiwbiAj4ntOkzSZGX4ZG/Ux0v+1o1vvVJrYM=;
        b=fXK2Y/P9RYah5H5Usg5XD7ROW3+c224PC1hsDZg5WY20szqAyp53YtvBheJC5PVPF5TsU5
        hgY7bylhkWSMTg6H6RbwlkuzzflSaZqnLEuT/RatR7SE3Y5lPRN6LqyoXJp5TfYW/qwlwA
        gisZvPU/Vsr4RHGxWQ6Gvt0jGKpBSns=
Received: from mail-wm1-f72.google.com (mail-wm1-f72.google.com
 [209.85.128.72]) (Using TLS) by relay.mimecast.com with ESMTP id
 us-mta-233-0mxKsjtMPtWmK7kH_7XrTA-1; Thu, 06 May 2021 09:38:35 -0400
X-MC-Unique: 0mxKsjtMPtWmK7kH_7XrTA-1
Received: by mail-wm1-f72.google.com with SMTP id
 g17-20020a05600c0011b029014399f816a3so1337366wmc.7
        for <kvm@vger.kernel.org>; Thu, 06 May 2021 06:38:34 -0700 (PDT)
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20161025;
        h=x-gm-message-state:from:to:cc:subject:date:message-id:in-reply-to
         :references:mime-version:content-transfer-encoding;
        bh=k4PLS6ZKiwbiAj4ntOkzSZGX4ZG/Ux0v+1o1vvVJrYM=;
        b=i7fXXwD2HxZm4uG1vmc0UX61/1uOUR4nQiqi4Pu/vtNNJwAWlnL2f9ml7kqTFXdIMB
         juDax4y5Dusd4CRNvwR2zQmORUuFCUz13gtBgvDdEPHhEPtxLTTed1lzVG48aSAHbFdj
         5O0OzwPliiTiNjZDBc9rZfGvPnfeBuc8n4NTaOdvOl9/Kq8OTMt56oaPJFiVN9aL2QTi
         VeIf5DpQg2dKlacopho1c10h6dIMSEj31BZ54n4nD1SWl2GZ4qhXDTgHQ88ZakST15ZI
         9AmOKtoiBqPkz8miMS/3QhnNPdHdxz5Yv3cQnkPGbfcUNic0+igid2iBfXgvdcbBexPL
         Dp3Q==
X-Gm-Message-State: AOAM532+K/EyMFIWkllDv9693Yksfg9zseFK7kL52q8pRFBo77I46qhJ
        lQ5MX9fZ40szz/IXtxAi8rjNhMuTYBPr5d66TMjhxMU4XavXUD06MdlRVrLoMn8pTdskf1YrSm4
        lLiDIQ1grskpL
X-Received: by 2002:a5d:4707:: with SMTP id y7mr5273917wrq.137.1620308314031;
        Thu, 06 May 2021 06:38:34 -0700 (PDT)
X-Google-Smtp-Source: 
 ABdhPJw9OkIRdMCjajJ35Xqm95nwOIWcYI/GC6OCGDnowdoEynI2VdSlNOXJBaO3mbso8BQyxIbuGA==
X-Received: by 2002:a5d:4707:: with SMTP id y7mr5273889wrq.137.1620308313868;
        Thu, 06 May 2021 06:38:33 -0700 (PDT)
Received: from localhost.localdomain
 (astrasbourg-652-1-219-60.w90-40.abo.wanadoo.fr. [90.40.114.60])
        by smtp.gmail.com with ESMTPSA id
 j13sm4830339wrd.81.2021.05.06.06.38.32
        (version=TLS1_3 cipher=TLS_AES_256_GCM_SHA384 bits=256/256);
        Thu, 06 May 2021 06:38:33 -0700 (PDT)
From: =?utf-8?q?Philippe_Mathieu-Daud=C3=A9?= <philmd@redhat.com>
To: qemu-devel@nongnu.org
Cc: kvm@vger.kernel.org, qemu-ppc@nongnu.org, qemu-arm@nongnu.org,
 Gerd Hoffmann <kraxel@redhat.com>, Paolo Bonzini <pbonzini@redhat.com>,
	=?utf-8?q?Alex_Benn=C3=A9e?= <alex.bennee@linaro.org>, =?utf-8?q?Philippe_M?=
	=?utf-8?q?athieu-Daud=C3=A9?= <philmd@redhat.com>
Subject: [PATCH v2 7/9] gdbstub: Replace alloca() + memset(0) by g_new0()
Date: Thu,  6 May 2021 15:37:56 +0200
Message-Id: <20210506133758.1749233-8-philmd@redhat.com>
X-Mailer: git-send-email 2.26.3
In-Reply-To: <20210506133758.1749233-1-philmd@redhat.com>
References: <20210506133758.1749233-1-philmd@redhat.com>
MIME-Version: 1.0
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org

The ALLOCA(3) man-page mentions its "use is discouraged".

Replace the alloca() and memset(0) calls by g_new0().

Signed-off-by: Philippe Mathieu-Daudé <philmd@redhat.com>
---
 gdbstub.c | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/gdbstub.c b/gdbstub.c
index 7cee2fb0f1f..666053bf590 100644
--- a/gdbstub.c
+++ b/gdbstub.c
@@ -1487,14 +1487,13 @@ static int process_string_cmd(void *user_ctx, const char *data,
         if (cmd->schema) {
             int schema_len = strlen(cmd->schema);
             int max_num_params = schema_len / 2;
+            g_autofree GdbCmdVariant *params = NULL;
 
             if (schema_len % 2) {
                 return -2;
             }
 
-            gdb_ctx.params = (GdbCmdVariant *)alloca(sizeof(*gdb_ctx.params)
-                                                     * max_num_params);
-            memset(gdb_ctx.params, 0, sizeof(*gdb_ctx.params) * max_num_params);
+            gdb_ctx.params = params = g_new0(GdbCmdVariant, max_num_params);
 
             if (cmd_parse_params(&data[strlen(cmd->cmd)], cmd->schema,
                                  gdb_ctx.params, &gdb_ctx.num_params)) {

From patchwork Thu May  6 13:37:57 2021
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 8bit
X-Patchwork-Submitter: =?utf-8?q?Philippe_Mathieu-Daud=C3=A9?=
 <philmd@redhat.com>
X-Patchwork-Id: 12242159
Return-Path: <kvm-owner@kernel.org>
X-Spam-Checker-Version: SpamAssassin 3.4.0 (2014-02-07) on
	aws-us-west-2-korg-lkml-1.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No, score=-19.4 required=3.0 tests=BAYES_00,DKIMWL_WL_HIGH,
	DKIM_SIGNED,DKIM_VALID,DKIM_VALID_AU,HEADER_FROM_DIFFERENT_DOMAINS,
	INCLUDES_CR_TRAILER,INCLUDES_PATCH,MAILING_LIST_MULTI,SPF_HELO_NONE,SPF_PASS,
	USER_AGENT_GIT autolearn=ham autolearn_force=no version=3.4.0
Received: from mail.kernel.org (mail.kernel.org [198.145.29.99])
	by smtp.lore.kernel.org (Postfix) with ESMTP id 96589C433ED
	for <kvm@archiver.kernel.org>; Thu,  6 May 2021 13:38:43 +0000 (UTC)
Received: from vger.kernel.org (vger.kernel.org [23.128.96.18])
	by mail.kernel.org (Postfix) with ESMTP id 62878613F3
	for <kvm@archiver.kernel.org>; Thu,  6 May 2021 13:38:43 +0000 (UTC)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S234369AbhEFNjk (ORCPT <rfc822;kvm@archiver.kernel.org>);
        Thu, 6 May 2021 09:39:40 -0400
Received: from us-smtp-delivery-124.mimecast.com ([216.205.24.124]:55674 "EHLO
        us-smtp-delivery-124.mimecast.com" rhost-flags-OK-OK-OK-OK)
        by vger.kernel.org with ESMTP id S233954AbhEFNjk (ORCPT
        <rfc822;kvm@vger.kernel.org>); Thu, 6 May 2021 09:39:40 -0400
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=redhat.com;
        s=mimecast20190719; t=1620308321;
        h=from:from:reply-to:subject:subject:date:date:message-id:message-id:
         to:to:cc:cc:mime-version:mime-version:content-type:content-type:
         content-transfer-encoding:content-transfer-encoding:
         in-reply-to:in-reply-to:references:references;
        bh=NaJZZTBiusehu1Tq7zBJHOToPc0jvSFCIFWhGsqf9q0=;
        b=fresYRepUTg/YtxMS/3eGVjkRpuHC2Xi5fEGESlljRXmD5PL2fWTexa6z2dHsysbpno9TJ
        tkwTRHAiN4vJN9wgANKJgwRKXkgLVDloiLlGdGf+yccTbZe6atcZxHWkoBcfxYN1bgt7Gi
        Pwe4Jhb+5LbDNGRhRbmSjgWJlSTV28w=
Received: from mail-wr1-f72.google.com (mail-wr1-f72.google.com
 [209.85.221.72]) (Using TLS) by relay.mimecast.com with ESMTP id
 us-mta-551-FEZgIZ2GPxqIYUxGuRdjdg-1; Thu, 06 May 2021 09:38:40 -0400
X-MC-Unique: FEZgIZ2GPxqIYUxGuRdjdg-1
Received: by mail-wr1-f72.google.com with SMTP id
 91-20020adf94640000b029010b019075afso2201587wrq.17
        for <kvm@vger.kernel.org>; Thu, 06 May 2021 06:38:39 -0700 (PDT)
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20161025;
        h=x-gm-message-state:from:to:cc:subject:date:message-id:in-reply-to
         :references:mime-version:content-transfer-encoding;
        bh=NaJZZTBiusehu1Tq7zBJHOToPc0jvSFCIFWhGsqf9q0=;
        b=CnKheQNPpcdngIDsD1ZZnP6FY5frkj/QxF5qjxecLNPD3jeoXekDJBnTjP1cZ0/+Pz
         i393J7b3h/ul9rFNfUgKvaLG6aAb5lz/yBbKB7Jxm11UJFEW0VXEX7c2CFSHckwmMrrL
         bm8KNm5R4+qL8XcciA3wJz9YLVMqpo6bDD60xeL8GgfnSAWFpkiSwU82IQKlqdZnEU8h
         zd/QcBUtcyyUkfBjlNcmeUzPUgxaYQrF6HS7n4S/FTLuP/Vn5Pk0IUomffceIRzY3GWa
         NYVHW0HzCqmdXArCgz826zhzESUm/p8qOT0wGsPZaRYQIG+N7uHTaZ3w0yrug68Qo6vs
         oPPw==
X-Gm-Message-State: AOAM531qT/7XSEIfhxFM5k+N2m9ckb9dpWVAh5OK//FJFeD3yygJ84S+
        6HrxqdR5+YLxnwlhUBxjPfqCMrx9GF1In2fvmAOZTm2phd75nZcNfJqqbugyTvwsmoDYUws1nGF
        MdruQ2o3CyFHP
X-Received: by 2002:adf:c002:: with SMTP id z2mr5292551wre.100.1620308318927;
        Thu, 06 May 2021 06:38:38 -0700 (PDT)
X-Google-Smtp-Source: 
 ABdhPJyyHdfkaTmkvWQf5Hf6Mx8u2g2OWRDThcbJG1gkPh5MWHU1eQptHFzQJqG0ffuS/NnFVO+6Rg==
X-Received: by 2002:adf:c002:: with SMTP id z2mr5292527wre.100.1620308318789;
        Thu, 06 May 2021 06:38:38 -0700 (PDT)
Received: from localhost.localdomain
 (astrasbourg-652-1-219-60.w90-40.abo.wanadoo.fr. [90.40.114.60])
        by smtp.gmail.com with ESMTPSA id
 l22sm9501029wmq.28.2021.05.06.06.38.37
        (version=TLS1_3 cipher=TLS_AES_256_GCM_SHA384 bits=256/256);
        Thu, 06 May 2021 06:38:38 -0700 (PDT)
From: =?utf-8?q?Philippe_Mathieu-Daud=C3=A9?= <philmd@redhat.com>
To: qemu-devel@nongnu.org
Cc: kvm@vger.kernel.org, qemu-ppc@nongnu.org, qemu-arm@nongnu.org,
 Gerd Hoffmann <kraxel@redhat.com>, Paolo Bonzini <pbonzini@redhat.com>,
	=?utf-8?q?Alex_Benn=C3=A9e?= <alex.bennee@linaro.org>, =?utf-8?q?Philippe_M?=
	=?utf-8?q?athieu-Daud=C3=A9?= <philmd@redhat.com>, =?utf-8?q?C=C3=A9dric_Le?=
	=?utf-8?q?_Goater?= <clg@kaod.org>, Peter Maydell <peter.maydell@linaro.org>,
 Andrew Jeffery <andrew@aj.id.au>, Joel Stanley <joel@jms.id.au>
Subject: [PATCH v2 8/9] hw/misc/pca9552: Replace g_newa() by g_new()
Date: Thu,  6 May 2021 15:37:57 +0200
Message-Id: <20210506133758.1749233-9-philmd@redhat.com>
X-Mailer: git-send-email 2.26.3
In-Reply-To: <20210506133758.1749233-1-philmd@redhat.com>
References: <20210506133758.1749233-1-philmd@redhat.com>
MIME-Version: 1.0
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org

The ALLOCA(3) man-page mentions its "use is discouraged".

Replace the g_newa() call by g_new().

Signed-off-by: Philippe Mathieu-Daudé <philmd@redhat.com>
---
 hw/misc/pca9552.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/hw/misc/pca9552.c b/hw/misc/pca9552.c
index b7686e27d7f..facf103cbfb 100644
--- a/hw/misc/pca9552.c
+++ b/hw/misc/pca9552.c
@@ -71,7 +71,7 @@ static void pca955x_display_pins_status(PCA955xState *s,
         return;
     }
     if (trace_event_get_state_backends(TRACE_PCA955X_GPIO_STATUS)) {
-        char *buf = g_newa(char, k->pin_count + 1);
+        g_autofree char *buf = g_new(char, k->pin_count + 1);
 
         for (i = 0; i < k->pin_count; i++) {
             if (extract32(pins_status, i, 1)) {

From patchwork Thu May  6 13:37:58 2021
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 8bit
X-Patchwork-Submitter: =?utf-8?q?Philippe_Mathieu-Daud=C3=A9?=
 <philmd@redhat.com>
X-Patchwork-Id: 12242161
Return-Path: <kvm-owner@kernel.org>
X-Spam-Checker-Version: SpamAssassin 3.4.0 (2014-02-07) on
	aws-us-west-2-korg-lkml-1.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No, score=-19.4 required=3.0 tests=BAYES_00,DKIMWL_WL_HIGH,
	DKIM_SIGNED,DKIM_VALID,DKIM_VALID_AU,HEADER_FROM_DIFFERENT_DOMAINS,
	INCLUDES_CR_TRAILER,INCLUDES_PATCH,MAILING_LIST_MULTI,SPF_HELO_NONE,SPF_PASS,
	USER_AGENT_GIT autolearn=ham autolearn_force=no version=3.4.0
Received: from mail.kernel.org (mail.kernel.org [198.145.29.99])
	by smtp.lore.kernel.org (Postfix) with ESMTP id 89214C433B4
	for <kvm@archiver.kernel.org>; Thu,  6 May 2021 13:38:50 +0000 (UTC)
Received: from vger.kernel.org (vger.kernel.org [23.128.96.18])
	by mail.kernel.org (Postfix) with ESMTP id 51340613EC
	for <kvm@archiver.kernel.org>; Thu,  6 May 2021 13:38:50 +0000 (UTC)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S234390AbhEFNjr (ORCPT <rfc822;kvm@archiver.kernel.org>);
        Thu, 6 May 2021 09:39:47 -0400
Received: from us-smtp-delivery-124.mimecast.com ([216.205.24.124]:45660 "EHLO
        us-smtp-delivery-124.mimecast.com" rhost-flags-OK-OK-OK-OK)
        by vger.kernel.org with ESMTP id S233954AbhEFNjq (ORCPT
        <rfc822;kvm@vger.kernel.org>); Thu, 6 May 2021 09:39:46 -0400
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=redhat.com;
        s=mimecast20190719; t=1620308327;
        h=from:from:reply-to:subject:subject:date:date:message-id:message-id:
         to:to:cc:cc:mime-version:mime-version:content-type:content-type:
         content-transfer-encoding:content-transfer-encoding:
         in-reply-to:in-reply-to:references:references;
        bh=RLFve6Y/zmLo3HdgEB/9T6FDXcQ0p9jwbmagngeIlOE=;
        b=XSZTY0dF3//VL+zAfbQ7M4B9+MMOYKVm8Ei1drOjWr40BijNSmiPlRFRYBzy5Pq/7KbvWn
        CQWJjFhTEYHrHD4iy1aBMt9uS5dxzBxxvEhY2u8zHx17IKh/znlgDNoIZ/l1TNNUZuash4
        c30fibj6K1qX1yOcNrtDZN75KSQDuo8=
Received: from mail-wm1-f72.google.com (mail-wm1-f72.google.com
 [209.85.128.72]) (Using TLS) by relay.mimecast.com with ESMTP id
 us-mta-275-uCjJ-GfmPE-8MebAzoRaXw-1; Thu, 06 May 2021 09:38:45 -0400
X-MC-Unique: uCjJ-GfmPE-8MebAzoRaXw-1
Received: by mail-wm1-f72.google.com with SMTP id
 y193-20020a1c32ca0000b029014cbf30c3f2so2646077wmy.1
        for <kvm@vger.kernel.org>; Thu, 06 May 2021 06:38:44 -0700 (PDT)
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20161025;
        h=x-gm-message-state:from:to:cc:subject:date:message-id:in-reply-to
         :references:mime-version:content-transfer-encoding;
        bh=RLFve6Y/zmLo3HdgEB/9T6FDXcQ0p9jwbmagngeIlOE=;
        b=fYBiqcTTNJ8o+5yN8rsXcGT5dUrf+hCB0kQFyOJDDF/NsRT23UdrwEHAy5+SxRLf6s
         LhD0mqW0sq388hwIYQXg6Vz/nCEQ/0e/Ugp6RT+WWk9PrHZ+W8TFg/R5tj0oXR+ntHRX
         BCiHscDb65m3vfTCb62X9TVlPrT7p7e5eiLuuTIuzPUqKHaom8W3B4Xq0xE0nm2I2KaC
         Ybfz4RGMgx9vGY0EabXR79Npu0FAe2IoA8IdCBDvSxX/0tdbLDYiTAhs6o1coLYyhoWh
         fG9NmjYkiWy18VRBR7P64FlJ0ZJZU4rWftFZgzDoD4N/6pD1aBOxTNBtbZTuNRW8vxmD
         gKOA==
X-Gm-Message-State: AOAM5332CF0hy/xO2QL1ghCQDQfUIgrdlSFE+8ACKy/Rg/hbgnOrJHZV
        iA0Rzcm5iqAv4IWyb0C8hI6Qt7Fu49NWvIrEaEALaHT1t5toM9SvkaYTOJkKqU5vu1lMmKEUx/L
        2LRLkXxOue8sA
X-Received: by 2002:adf:e348:: with SMTP id n8mr5215091wrj.209.1620308324012;
        Thu, 06 May 2021 06:38:44 -0700 (PDT)
X-Google-Smtp-Source: 
 ABdhPJwvsmJHX73t3ak/OC4xbrrBRJCZcdb4T22RK9A3SSLmm4vJNsQC6Ff+qwwaWbT6aA16s1zISA==
X-Received: by 2002:adf:e348:: with SMTP id n8mr5215061wrj.209.1620308323810;
        Thu, 06 May 2021 06:38:43 -0700 (PDT)
Received: from localhost.localdomain
 (astrasbourg-652-1-219-60.w90-40.abo.wanadoo.fr. [90.40.114.60])
        by smtp.gmail.com with ESMTPSA id
 w25sm3208909wmk.39.2021.05.06.06.38.42
        (version=TLS1_3 cipher=TLS_AES_256_GCM_SHA384 bits=256/256);
        Thu, 06 May 2021 06:38:43 -0700 (PDT)
From: =?utf-8?q?Philippe_Mathieu-Daud=C3=A9?= <philmd@redhat.com>
To: qemu-devel@nongnu.org
Cc: kvm@vger.kernel.org, qemu-ppc@nongnu.org, qemu-arm@nongnu.org,
 Gerd Hoffmann <kraxel@redhat.com>, Paolo Bonzini <pbonzini@redhat.com>,
	=?utf-8?q?Alex_Benn=C3=A9e?= <alex.bennee@linaro.org>, =?utf-8?q?Philippe_M?=
	=?utf-8?q?athieu-Daud=C3=A9?= <philmd@redhat.com>,
 David Gibson <david@gibson.dropbear.id.au>, Greg Kurz <groug@kaod.org>
Subject: [PATCH v2 9/9] target/ppc/kvm: Replace alloca() by g_malloc()
Date: Thu,  6 May 2021 15:37:58 +0200
Message-Id: <20210506133758.1749233-10-philmd@redhat.com>
X-Mailer: git-send-email 2.26.3
In-Reply-To: <20210506133758.1749233-1-philmd@redhat.com>
References: <20210506133758.1749233-1-philmd@redhat.com>
MIME-Version: 1.0
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org

The ALLOCA(3) man-page mentions its "use is discouraged".

Replace it by a g_malloc() call.

Signed-off-by: Philippe Mathieu-Daudé <philmd@redhat.com>
Reviewed-by: Greg Kurz <groug@kaod.org>
---
 target/ppc/kvm.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/target/ppc/kvm.c b/target/ppc/kvm.c
index 104a308abb5..63c458e2211 100644
--- a/target/ppc/kvm.c
+++ b/target/ppc/kvm.c
@@ -2698,11 +2698,10 @@ int kvmppc_save_htab(QEMUFile *f, int fd, size_t bufsize, int64_t max_ns)
 int kvmppc_load_htab_chunk(QEMUFile *f, int fd, uint32_t index,
                            uint16_t n_valid, uint16_t n_invalid, Error **errp)
 {
-    struct kvm_get_htab_header *buf;
     size_t chunksize = sizeof(*buf) + n_valid * HASH_PTE_SIZE_64;
+    g_autofree struct kvm_get_htab_header *buf = g_malloc(chunksize);
     ssize_t rc;
 
-    buf = alloca(chunksize);
     buf->index = index;
     buf->n_valid = n_valid;
     buf->n_invalid = n_invalid;
